src/core/data/store.cc

```diff
@@ -136,8 +136,9 @@ FutureWrapper::FutureWrapper(
 #ifdef DEBUG_LEGATE
     assert(!initialize || future_.get_untyped_size() == field_size);
 #endif
-    auto proc     = Processor::get_executing_processor();
-    auto mem_kind = proc.kind() == Processor::Kind::TOC_PROC ? Memory::Kind::GPU_FB_MEM
+    auto proc = Processor::get_executing_processor();
+    // TODO: 2022-10-04: Work around a Legion bug, by not instantiating futures on framebuffer.
+    auto mem_kind = proc.kind() == Processor::Kind::TOC_PROC ? Memory::Kind::Z_COPY_MEM
```

Comment:
you can use this function: https://github.com/nv-legate/legate.core/blob/branch-22.12/src/core/utilities/machine.cc#L23-L28

Reply:
fixed